http/src/test/scala/org/http4s/blaze/http/http2/ServerPriorKnowledgeHandshakerSpec.scala

```diff
@@ -0,0 +1,116 @@
+package org.http4s.blaze.http.http2
+
+import org.http4s.blaze.http.http2.mocks.MockHeadStage
+import org.http4s.blaze.http.http2.server.ServerPriorKnowledgeHandshaker
+import org.http4s.blaze.pipeline.LeafBuilder
+import org.specs2.mutable.Specification
+
+import scala.util.{Failure, Success}
+
+class ServerPriorKnowledgeHandshakerSpec extends Specification {
+
+  private def makeHandshaker(localSettings: ImmutableHttp2Settings): ServerPriorKnowledgeHandshaker = {
+    val flowStrategy = new DefaultFlowStrategy(localSettings)
+    new ServerPriorKnowledgeHandshaker(
+      localSettings = localSettings,
+      flowStrategy = flowStrategy,
+      nodeBuilder = _ => ???
+    )
+  }
+
+  "ServerPriorKnowledgeHandshaker">> {
+    "can perform a handshake" in {
+      val localSettings = Http2Settings.default
+      val head = new MockHeadStage
+      val handshaker = makeHandshaker(localSettings)
+      LeafBuilder(handshaker).base(head)
+
+      val serverSession = handshaker.handshake()
+      head.reads.dequeue().success(bits.getPrefaceBuffer())
+
+      // Consume all the inbound data which may be a prelude and the local settings
+      head.consumeOutboundData()
+      head.consumeOutboundData()
```

Comment:
This is called twice for prelude and local settings separately?

Reply:
Correct. I'll add a note to make that clear because it does look suspect.